### PR TITLE
Fix build for MacOS

### PR DESCRIPTION
**Pull request by rkimballn1**
Builds and unit tests pass for CPU and INTERPRETER backends. Did not test GPU on macos because I can't.

The change in loop_kernel.?pp was because m_outputs shadows the variable of the same name in class Node and I was getting a compiler warning.

### CODE REVIEW

File: src/ngraph/codegen/file_util.cpp [hunk 1]
```diff
@@ -0,0 +1,349 @@
...
+        path = "/tmp";
+    }
+
+    return path;
+}
+
+vector<char> codegen::file_util::read_file_contents(const string& path)
+{
+    size_t file_size = get_file_size(path);
+    vector<char> data;
```
Comment (megawidget) [inline src/ngraph/codegen/file_util.cpp]: why not just `vector<char> data(file_size);`?

File: src/ngraph/codegen/file_util.cpp [hunk 2]
```diff
@@ -0,0 +1,349 @@
...
+            case DT_REG:
+            {
+                string file_name = file_util::path_join(path, name);
+                func(file_name, false);
+                break;
+            }
+            default: break;
+            }
+        }
+        closedir(dir);
```
Comment (megawidget) [inline src/ngraph/codegen/file_util.cpp]: what if `func` throws? you'll leak the pointer?  should `func` be `noexcept`?

File: src/ngraph/codegen/file_util.hpp [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+namespace ngraph
+{
+    namespace codegen
+    {
+        class file_util;
+    }
+}
+
+class ngraph::codegen::file_util
```
Comment (megawidget) [inline src/ngraph/codegen/file_util.hpp]: i'm a bit confused why this is a class and not a namespace, but :okay:

File: src/ngraph/codegen/compiler.cpp [hunk 4]
```diff
@@ -260,6 +278,21 @@ bool codegen::StaticCompiler::is_version_number(const string& path)
...
 
+static template <typename U, typename T>
+bool contains(const U& container, const T& obj)
+{
+    bool rc = false;
+    for (auto o : container)
+    {
+        if (o == obj)
+        {
+            rc = true;
```
[the remark this answers is not in the document]
Comment (rkimballn1) [reply src/ngraph/codegen/compiler.cpp]: Yes, but I believe in single return value.

File: src/ngraph/codegen/file_util.cpp (hunk 2 above)
Comment (rkimballn1) [reply src/ngraph/codegen/file_util.cpp]: My understanding of noexcept is that it is meaningless. A noexcept function can throw exceptions.

Comment (megawidget) [reply src/ngraph/codegen/file_util.cpp]: ok, so it needs a guard then?

**Comment by avijit-nervana** [comment]
Replacing myself with @diyessi so that Scott can assign someone from the nGraph core team.

File: src/ngraph/codegen/compiler.cpp (hunk 4 above)
[the remark this answers is not in the document]
Comment (diyessi) [reply src/ngraph/codegen/compiler.cpp]: @megawidget 's is simpler, but even simpler is
`return std::find(container.begin(), container.end(), obj) != container.end()`

File: src/ngraph/codegen/file_util.cpp (hunk 2 above)
Comment (diyessi) [reply src/ngraph/codegen/file_util.cpp]: yes it does.

File: src/ngraph/codegen/file_util.cpp [hunk 5]
```diff
@@ -0,0 +1,341 @@
...
+        rc = tmpname;
+    }
+
+    free(tmpname);
+    return rc;
+}
+
+std::string codegen::file_util::get_temp_directory()
+{
+    const vector<string> potential_tmps = {"NGRAPH_TMP", "TMPDIR", "TMP", "TEMP", "TEMPDIR"};
```
Comment (diyessi) [inline src/ngraph/codegen/file_util.cpp]: This list seems like overkill.

File: src/ngraph/codegen/file_util.cpp [hunk 6]
```diff
@@ -0,0 +1,341 @@
...
+    if (mkdtemp(tmpname))
+    {
+        rc = tmpname;
+    }
+
+    free(tmpname);
+    return rc;
+}
+
+std::string codegen::file_util::get_temp_directory()
```
Comment (diyessi) [inline src/ngraph/codegen/file_util.cpp]: I think `get_temp_directory_path` is a bette name.

File: src/ngraph/codegen/file_util.cpp [hunk 7]
```diff
@@ -0,0 +1,341 @@
...
+    {
+        path = getenv(var.c_str());
+        if (path != nullptr)
+        {
+            break;
+        }
+    }
+    if (path == nullptr)
+    {
+        path = "/tmp";
```
Comment (diyessi) [inline src/ngraph/codegen/file_util.cpp]: This won't work on native windows will it?

File: src/ngraph/codegen/file_util.hpp (hunk 3 above)
Comment (diyessi) [reply src/ngraph/codegen/file_util.hpp]: I agree -- it should be a namespace.

File: src/ngraph/codegen/file_util.hpp [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+{
+public:
+    static std::string get_file_name(const std::string&);
```
Comment (diyessi) [inline src/ngraph/codegen/file_util.hpp]: These should all have short doc strings.

File: src/ngraph/codegen/compiler.cpp (hunk 4 above)
[the remark this answers is not in the document]
Comment (rkimballn1) [reply src/ngraph/codegen/compiler.cpp]: I removed this function as it exists elsewhere

[the remark this answers is not in the document]
Comment (rkimballn1) [reply src/ngraph/codegen/compiler.cpp]: Also @diyessi your solution does not work for all instances of the templated function, it does not work with unordered_set. Less code is not always better nor more readable.

File: src/ngraph/codegen/file_util.cpp (hunk 5 above)
Comment (rkimballn1) [reply src/ngraph/codegen/file_util.cpp]: This list is to cover multiple operating systems. I would rather have too many than too few

File: src/ngraph/codegen/file_util.cpp (hunk 7 above)
Comment (rkimballn1) [reply src/ngraph/codegen/file_util.cpp]: Windows defines both TEMP and TMP so the environment variable defining the temp directory will be found before we get here.

**Comment by rkimballn1** [comment]
TESTNOW

**Pull request merged**